pyart/io/output_to_geotiff.py

```diff
@@ -0,0 +1,318 @@
+    The former will require an SLD file for colorful display using standard
+    GIS or web mapping software, while the latter will show colors
+    "out-of-the-box" but lack actual data values. The function also can
+    output an SLD file based on the user-specified inputs. User can specify
+    the 2D vertical level to be output. If this is not specified, a 2D
+    composite is created. User also can specify the field to output.
+
+    This function requires GDAL Python libraries to be installed. These are
+    available via conda; e.g., 'conda install gdal'
+
+    Parameters
+    ----------
+    grid : pyart.core.Grid object
+        Grid object to write to file.
+    filename : str
+        Filename template for the GeoTIFF. The field name will be appended
+        prior to the suffix. For example, if the filename template given is
+        'radar.tif', and the field name is 'reflectivity', then
+        the final file will be called 'radar_reflectivity.tif'.
+    field : str
+        Field name to output to file. This also will be used to create the
+        final GeoTIFF filename.
+
+    Other Parameters
+    ----------------
+    rbg : bool, optional
+        True - Output 3-band RGB GeoTIFF
+        False - Output single-channel, float-valued GeoTIFF. For display,
+                likely will need an SLD file to provide a color table.
+    level: int or None, optional
+        Index for z-axis plane to output. None gives composite values
+        (i.e., max in each vertical column).
+    cmap : str or matplotlib.colors.Colormap object, optional
+        Colormap to use for RGB output or SLD file.
+    vmin : int or float, optional
+        Minimum value to color for RGB output or SLD file.
+    vmax : int or float, optional
+        Maximum value to color for RGB output or SLD file.
+    color_levels : int or None, optional
+        Number of color levels in cmap. Useful for categorical colormaps
+        with steps << 255 (e.g., hydrometeor ID).
+    warp : bool, optional
+        True - Use gdalwarp (called from command line using os.system)
+               to warp to a lat/lon WGS84 grid.
+        False - No warping will be performed. Output will be Az. Equidistant.
+    sld : bool, optional
+        True - Create a Style Layer Descriptor file (SLD) mapped to vmin/vmax
+               and cmap. File is named same as output TIFF, except for .sld
+               extension.
+        False - Don't do this.
+
+    Returns
+    -------
+    None
+
+    """
+    if not IMPORT_FLAG:
+        raise MissingOptionalDependency(
+            'GDAL not detected, GeoTIFF output failure!')
+
+    if field in grid.fields.keys():
+        # Determine whether filename template already contains a suffix
+        # If not, append an appropriate one.
+        if '.' in filename:
+            name, end = filename.split('.')
+        else:
+            name = filename
+            end = 'tif'
+        ofile = name + "_" + field + "." + end
```

Comment:
Changed to not use the field in the filename. If there is not a suffix, however, '.tif' is appended so that `_create_sld` gets a suffix it can change to '.sld'